src/api/gamestate.cc

```diff
@@ -0,0 +1,50 @@
+#include "gamestate.h"
+#include <utility>
+#include <vector>
+#include "../statefunctions/statecontroller.h"
+#include "types.h"
+
+mahjong::GameSettings convertGameSettings(const CGameSettings* settings) {
+  mahjong::GameSettings cpp_settings;
+
+  cpp_settings.seed = settings->seed;
+
+  // Char arrays to vector
+  if (settings->seatControllers && settings->numControllers > 0) {
+    for (int i = 0; i < settings->numControllers; i++) {
+      if (settings->seatControllers[i]) {
+        cpp_settings.seatControllers.emplace_back(settings->seatControllers[i]);
+      }
+    }
+  }
+
+  return cpp_settings;
+}
+
+extern "C" {
+
+int StartGame(const CGameSettings* settings, int async) {
+  const mahjong::GameSettings cpp_settings = convertGameSettings(settings);
+  return mahjong::StartGame(cpp_settings, async != 0);
+}
+
+void ExitGame(int game) {
+  mahjong::ExitGame(game);
+}
+
+CGameState* InitGameState(const CGameSettings* settings) {
+  const mahjong::GameSettings cpp_settings = convertGameSettings(settings);
+  auto state = mahjong::InitGameState(cpp_settings);
+  return new CGameState{std::move(state)};
+}
+
+CGameState* AdvanceGameState(CGameState* state) {
```

Comment:
review to myself, I should add a test to init a game state and try advancing it one